internal/node/component/network/relation.go

```diff
@@ -0,0 +1,1836 @@
+package network
+
+import (
+	"github.com/rss3-network/node/schema/worker"
+	"github.com/rss3-network/protocol-go/schema/network"
+)
+
+type ConfigDetail struct {
+	IsRequired  bool        `json:"isRequired"`
+	Type        string      `json:"type"`
+	Value       interface{} `json:"value"`
+	Description string      `json:"description"`
+}
+
+type Endpoint struct {
+	URL           *ConfigDetail `json:"network"`
+	HTTPHeaders   *ConfigDetail `json:"http_headers"`
+	HTTP2Disabled *ConfigDetail `json:"http2_disabled"`
+}
+
+type Parameters struct {
+	BlockHeightStart      *ConfigDetail `json:"block_height_start,omitempty"`
+	BlockHeightTarget     *ConfigDetail `json:"block_height_target,omitempty"`
+	BlockNumberStart      *ConfigDetail `json:"block_number_start,omitempty"`
+	BlockNumberTarget     *ConfigDetail `json:"block_number_target,omitempty"`
+	RPCThreadBlocks       *ConfigDetail `json:"rpc_thread_blocks,omitempty"`
+	RPCBatchBlocks        *ConfigDetail `json:"rpc_batch_blocks,omitempty"`
+	RPCBatchReceipts      *ConfigDetail `json:"rpc_batch_receipts,omitempty"`
+	RPCBatchBlockReceipts *ConfigDetail `json:"rpc_batch_block_receipts,omitempty"`
+	APIKey                *ConfigDetail `json:"api_key,omitempty"`
+}
+
+type workerConfig struct {
+	WorkerID     ConfigDetail  `json:"worker_id"`
+	Network      ConfigDetail  `json:"network"`
+	Worker       ConfigDetail  `json:"worker"`
+	EndpointID   *ConfigDetail `json:"endpoint_id,omitempty"`
+	Endpoint     *Endpoint     `json:"endpoint,omitempty"`
```

Comment:
what is the relationship between Endpoint and EndpointID? If an Endpoint exists, should the URL in EndpointID be replaced?

Reply:
Fixed. If a global endpoint_id is defined in endpoints, we will use the pre-defined one. Otherwise, we will use it as a pure URL.

https://github.com/RSS3-Network/Node/blob/develop/config/config.go#L49

so the config center frontend should have a part to define all global endpoints so users can choose it?

https://github.com/RSS3-Network/Node/blob/develop/deploy/config.yaml#L46

Reply:
it shouldn't be included in each workers' configs.